### ISSUE TITLE
Cannot access reassignable field from inflight

### ISSUE THREAD

**Issue by Chriscbr** [winglang/wing#2770]
### I tried this:

```js
bring "@cdktf/provider-aws" as aws;

class MyTable {
  table: aws.dynamodbTable.DynamodbTable;
  init() {
    this.table = new aws.dynamodbTable.DynamodbTable();
  }

  inflight printName(): str {
    log(this.table.name);
  }
}
```

### This happened:

```
error: Cannot capture reassignable field 'name'
   --> access_reassignable.w:10:20
   |
10 |     log(this.table.name);
   |                    ^^^^ Cannot capture reassignable field 'name'
```

### I expected this:

It compiles

### Is there a workaround?

Copy the value into a dedicated field that is not reassignable

```js
bring "@cdktf/provider-aws" as aws;

class MyTable {
  table: aws.dynamodbTable.DynamodbTable;
  tableName: str;
  init() {
    this.table = new aws.dynamodbTable.DynamodbTable();
    this.tableName = this.table.name;
  }

  inflight printName(): str {
    log(this.tableName);
  }
}
```

### Component

Compiler

### Wing Version

0.18.7

### Node.js Version

18.14.1

### Platform(s)

MacOS

### Anything else?

Under the hood, the class `aws.dynamodbTable.DynamodbTable` that is imported from CDKTF is modeled in TypeScript like so:

```ts
export declare class DynamodbTable extends cdktf.TerraformResource {
    static readonly tfResourceType = "aws_dynamodb_table";
    constructor(scope: Construct, id: string, config: DynamodbTableConfig);
    private _name?;
    get name(): string;
    set name(value: string);
    // ... 100+ other fields and methods
}
```

Since there's a getter and setter, JSII models "name" as a single property, so when we're importing JSII information in the Wing compiler, the compiler treats the field as reassignable. (If there wasn't a setter, JSII would model "name" as a property with an "immutable" flag, and Wing would not interpret the field as reassignable).

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by Chriscbr** [comment]
I figured out a workaround and added it to the issue.

I guess the existing behavior is legitimate so I'm not sure this deserves to be a bug, but on the other hand the workaround isn't necessarily intuitive (I encountered it yesterday and got stumped so I had to revisit it today). I wonder if we should surface the workaround using a better error message or in our docs?

Related: https://github.com/winglang/wing/discussions/1965
